### PR TITLE
FEATURE: allow to select nodetype

### PR DESCRIPTION
**Pull request by dimaip**

### CODE REVIEW

File: Configuration/Settings.yaml [hunk 1]
```diff
@@ -19,6 +19,7 @@ Neos:
             #   type: flat
             #   query: 'Search.query(node).nodeType("Your.Namespace:News").sortDesc("date").from((page - 1) * 20).limit(20).execute().toArray()'
             #   newReferenceNodePath: '/sites/site'
+            #   # If now `newNodeType` is defined, then the nodetype selection dialog would be shown
```
Comment (DrillSergeant) [inline Configuration/Settings.yaml]: Typo: "now" should be "no"

**Comment by DrillSergeant** [comment]
I seems this happens when I did not select a node before in the tree. When I select a node before (does not matter with one), I get the insert-dialog. 

But it is not in the "into-mode" of the parent node:
image
Right now it shows the "after-mode" of the parent-node.

**Comment by dimaip** [comment]
Hmm this is really strange... Are you sure you have `newReferenceNodePath` set correctly? I can not reproduce it on my setup.

Regarding the insert mode, unfortunately it's not possible to influence it from the outside, without overwriting the whole AddNode dialog, but it's possible to limit creation of adjacent nodes with constraints. Is it not an option for your usecase?

**Comment by dimaip** [comment]
I guess if we must do it, it would require a core intervention, we should make it possible to set the desired insert mode from the redux action, but that again would require ~1-2h of work.

**Comment by DrillSergeant** [comment]
**Empty dialog**
I can reproduce "the empty dialog box" if I log out of the neos backend, log back in and directly click the plus-icon (in my custom TreeNav-Tab). Import is to not select any node before. After I click any node, the add-button works. Can you reproduce it with those steps?

**Insert mode**
I think this is really needed to land in the correct mode. Editors are not aware of this and they will complain that something is not working. We can sponsor the additional 1-2h.

**Comment by DrillSergeant** [comment]
@dimaip , I just tested again and waited really long before clicking the "+"-button. Same issue still. For me it looks like the FlatTree does not have an active document-node. Once I click any document-node it works.

**Pull request merged**